### PR TITLE
fix: react-combobox check icon slot does not collapse

### LINKED ISSUE #32227: [Bug]: Dropdown checkbox squished when unchecked and string is longer than Dropdown width
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Linux 6.5 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (32) x64 AMD EPYC 7763 64-Core Processor
    Memory: 89.58 GB / 125.78 GB
    Container: Yes
    Shell: 5.8.1 - /bin/zsh
```

### Are you reporting an Accessibility issue?

no

### Reproduction

https://stackblitz.com/edit/uyeyc5-j3jf4v?file=src%2Fexample.tsx

### Bug Description

When a Dropdown contains checkboxes and the checkbox is unchecked for an entry that has a string longer than the Dropdown width, the checkbox renders squished.

Internal repro: Outlook Monarch > Settings > General > Notifications > expand top section. See screenshot below.

## Actual Behavior
Squished checkbox

## Expected Behavior
Normal, unchecked checkbox

image

### Logs

_No response_

### Requested priority

High

### Products/sites affected

Outlook Monarch

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by smhigley**
Adds one style fix and one story.

## Previous Behavior

When the text of an option was longer than could fit in the available space, the check icon would collapse:
screenshot showing the check icon appearing as a single vertical line

## New Behavior

Fixes the icon slot collapsing with `flex-shrink: 0`, and adds a story showing how to add `overflow-wrap` styles to option text.

## Related Issue(s)

- Fixes #32227 and #32228

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `103.763 kB`<br />`34.054 kB` | `103.808 kB`<br />`34.08 kB` | `45 B` <br />`26 B` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `104.349 kB`<br />`33.963 kB` | `104.394 kB`<br />`33.99 kB` | `45 B` <br />`27 B` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.096 MB`<br />`270.653 kB` | `1.096 MB`<br />`270.671 kB` | `45 B` <br />`18 B` |
| <samp>react-tag-picker</samp> <br /> <abbr title='bundle-size/Package.fixture.js'>@fluentui/react-tag-picker - package</abbr> | `181.367 kB`<br />`54.538 kB` | `181.412 kB`<br />`54.567 kB` | `45 B` <br />`29 B` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `106.751 kB`<br />`35.596 kB` | `106.796 kB`<br />`35.626 kB` | `45 B` <br />`30 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.141 kB`<br />`20.157 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `217.412 kB`<br />`63.063 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.442 kB`<br />`14.607 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/dc714f81be7ddeefba0c26ca18354959580fcd77'>dc714f81be7ddeefba0c26ca18354959580fcd77</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/Avatar-mount_base.html">672</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/Avatar-mount.html">626</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/Button-mount_base.html">317</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/Button-mount.html">290</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/Field-mount_base.html">1104</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/Field-mount.html">1169</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/FluentProvider-mount_base.html">707</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/FluentProvider-mount.html">731</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/FluentProviderWithTheme-mount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/FluentProviderWithTheme-mount.html">82</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">35</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">35</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">82</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/MakeStyles-mount_base.html">863</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/MakeStyles-mount.html">879</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/Persona-mount_base.html">1791</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/Persona-mount.html">1679</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/SpinButton-mount_base.html">1401</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/SpinButton-mount.html">1418</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/SwatchPicker-mount_base.html">1667</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32243/perf-test-react-components/SwatchPicker-mount.html">1715</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by khmakoto — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-combobox/stories/src/Dropdown/DropdownTruncation.stories.tsx [hunk 1]
```diff
@@ -35,14 +40,15 @@ export const TruncatedValue = (props: Partial<DropdownProps>) => {
     'Fox',
     'Hamster',
     'Snake',
+    'SuperLongName_123456789@soyouwantalongurl.com',
```
Comment (spmonahan) [inline packages/react-components/react-combobox/stories/src/Dropdown/DropdownTruncation.stories.tsx:43]: I think we should avoid adding URLs we don't own to our docs.

```suggestion
    'SuperLongName_123456789_SomeMoreStuffToMakeItLonger@fluentui.dev',
```

Comment (smhigley) [reply packages/react-components/react-combobox/stories/src/Dropdown/DropdownTruncation.stories.tsx:43]: oh good call :D

File: change/@fluentui-react-combobox-7d75d97b-877f-4f43-92d4-cfadb37677da.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-combobox-7d75d97b-877f-4f43-92d4-cfadb37677da.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32243&commitId=ceb5956b0aec71239ce64e56d342ccb36b89a678&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Basic High Contrast.chromium.png|2| Changed|
|Avatar Converged.Badge Mask RTL.chromium.png|2| Changed|
|Avatar Converged.badgeMask.chromium.png|11| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by spmonahan — APPROVED** [review]

**Pull request merged**